### PR TITLE
Enable pylint on Travis CI builds

### PR DESCRIPTION
**Pull request by karlding**
Run pylint on all Travis CI builds for rules that we currently are
conformant to.

This adds a .pylintrc-wip file, which is intended to be a temporary
stop-gap that contains the current rules that the project is conformant
to. This allows us to incrementally enable pylint rules and have them
them enforced in subsequent code, rather than fixing the violations all
at once.

The .pylintrc file is intended to be the file containing all the
preferred set of rules that the project will eventually be compliant
against.

Currently the `.pylintrc` files are generated via the default rules:

```bash
pylint --generate-rcfile >> .pylintrc
pylint --disable=all --generate-rcfile >| .pylintrc-wip
```

I don't really have a preference in terms of which linter is being
used, so let me know if you feel strongly about a particular linter.

### CODE REVIEW

File: .pylintrc [hunk 1]
```diff
@@ -0,0 +1,570 @@
...
+# regex matches against base names, not paths.
+ignore-patterns=
+
+# Python code to execute, usually for sys.path manipulation such as
+# pygtk.require().
+#init-hook=
+
+# Use multiple processes to speed up Pylint. Specifying 0 will auto-detect the
+# number of processors available to use.
+jobs=1
```
Comment (felixdivo) [inline .pylintrc]: Maybe use `0` since we might run it on our local multicore systems?

Comment (felixdivo) [reply .pylintrc]: Applies to both files.

**Comment by felixdivo** [comment]
This fixes #579, right?

**Comment by hardbyte** [comment]
This PR only addresses the linter aspect of #579.

File: .pylintrc (hunk 1 above)
Comment (karlding) [reply .pylintrc]: Sorry, I was out of town for the last few days. Do you still want me to change this?

Comment (felixdivo) [reply .pylintrc]: It's not too important. We can just do it next time we remove some other exclusions.

**Pull request merged**